### ISSUE TITLE
Inform user when she is blocked from editing Wikidata too

### ISSUE THREAD

**Issue by neslihanturan** [commons-app/apps-android-commons#1605]
**Summary:** 

Like we do that for Commons #1511 , we have to inform user when blocked from Wikidata too. Because on Nearby Activity we edit wikidata item. We can use same logic at #1573

**Comment by seannemann21** [comment]
OK I can take a look at this since I worked on the last blocking notification!

**Comment by seannemann21** [comment]
@neslihanturan they Wikidata:API page (https://www.wikidata.org/wiki/Wikidata:API) redirects me to the wikimedia page. Do you think that I can assume that the blocking mechanism for the two APIs function identically? Or do you think I should try to find someone to block me to check this just in case of a difference?

**Comment by seannemann21** [comment]
Also, since I can reuse code from #1573 should I just update that pull request so that it also fixes this issue? Or should I create a new branch off of master that does not have the code from #1573 and create a new pull request? If I did the latter, would there be a way to avoid rewriting the code from #1573? I'm slightly confused with how to proceed since #1573 is not merged into master yet. Thanks!

**Comment by nicolas-raoul** [comment]
We can probably assume it is the same, I would say.

You can create a sub-branch that starts from your latest commit on the 1573 branch, that would be the best to benefit from its code while not modifying it (modifying it would postpone its merge).

**Comment by seannemann21** [comment]
OK thank you very much! I just wasn't sure if a problem would arise if the pull request I made for this issue was accepted before the pull request in #1573, but I'll just go ahead and create a sub-branch!

**Comment by seannemann21** [comment]
OK everything seems to be working; however, it doesn't look like you ever sign into Wikidata within the app. As a result, the block notification only appears on the nearby screen if you previously went to wikidata.org and signed in since then your IP address would have been blocked. Is this expected behavior? In which case, I should submit a pull request since the blocking functionality would be working, correct? If not, let me know what I should do! Thank you!

**Comment by seannemann21** [comment]
Picking up on the IP address for the block status api call appears to be inconsistent. I should probably figure out how to log the user into wikidata

**Comment by seannemann21** [comment]
OK so I slightly modified the current login method to also login to wikidata. The blocking notification now works as expected. I only modified the login for testing purposes (and will revert the changes before I submit my pull request). Now my question is, should I submit a pull request for this issue since the blocking notification would work if the user was signed into Wikidata? Or should I make the necessary changes to also log the user into Wikidata as well (probably creating a separate issue and pull request) and then create this pull request? Sorry if this is a bit confusing! Any help would be greatly appreciated!

**Comment by nicolas-raoul** [comment]
How about logging in only when needed?
I expect that only like 10% of the users will use Wikidata-editing features, and most of them only like once per week even if they use the app everyday.
So logging into Wikidata at every start of the app sounds like it would hit performance more than needed.

**Comment by seannemann21** [comment]
Ok, yes that makes sense! Just wondering, is clicking on "Wikidata item" on this screen the only way to edit Wikidata from within the app?

![image](https://user-images.githubusercontent.com/32438079/41419380-dbed9f96-6fbf-11e8-8ff1-82f64caea2bd.png)

**Comment by seannemann21** [comment]
Also do you suggest that I open a login screen if the user clicks on "Wikidata item"? Although at this point, if the user clicks on "Wikidata item" and tries to edit something, then they will be informed already. Any advice would be greatly appreciated! Thank you, again!

![image](https://user-images.githubusercontent.com/32438079/41419712-aed7ccd8-6fc0-11e8-8f63-b1ced4603754.png)

**Comment by seannemann21** [comment]
Also, sorry for so many questions! This is mostly new to me

**Comment by misaochan** [comment]
@seannemann21 No worries! :) There is no way to manually edit Wikidata from within the app - it happens when you upload a picture to a Nearby item. E.g. in your screenshot, tap the picture icon and upload a picture, then later visit the Crescent Springs item on Wikidata and you should see that the picture has been added to P18.

**Comment by seannemann21** [comment]
So if a user can't manually edit Wikidata from within the app, what is the point of displaying a notification that they are blocked from Wikidata? Just trying to better understand the app

**Comment by nicolas-raoul** [comment]
When misaochan said that there is no way to **manually** edit Wikidata from within the app, the emphasize was on "manually", she meant that the only way is the automatic edit that happens when you follow the steps below:

1. In the left menu, touch "Nearby"
2. A map shows with several pins. Touch one of the pins near you
3. Press the "+" that appeared
4. Press the camera icon
5. Go take a picture of the thing
6. Fill metadata and categories, the picture gets uploaded to Commons
7. At this step, our app links the Commons picture from the Wikidata item. That's where the edit happens.

**Comment by seannemann21** [comment]
So if a user is blocked from wikidata, then will the user be blocked from uploading a picture in this fashion?

**Comment by seannemann21** [comment]
In which case should I inform the user that they are blocked when the user opens the nearby tab or when the user goes to upload a picture in the way that you stated? Additionally, since I need the user to log in to wikidata to determine if the user is blocked, do I present another log in screen?

**Comment by misaochan** [comment]
> So if a user is blocked from wikidata, then will the user be blocked from uploading a picture in this fashion?

No, the upload will proceed if they are blocked, but the P18 edit to the corresponding wikidata item won't happen.

> In which case should I inform the user that they are blocked when the user opens the nearby tab or when the user goes to upload a picture in the way that you stated? 

I believe the ideal way is when they upload the picture and the P18 edit doesn't go through.

> Additionally, since I need the user to log in to wikidata to determine if the user is blocked, do I present another log in screen?

AFAIK we already log them in to Wikidata automatically for the Nearby edits? Hopefully @maskaravivek can clarify.

**Comment by seannemann21** [comment]
Ok thank you! This clears things up! I was checking their block status as soon as they opened the nearby activity. They are probably logged in once they go to upload the picture, maybe

**Comment by seannemann21** [comment]
@maskaravivek do you know if/when the user is logged into wikidata? I've tried uploading a picture and then checking the block status, but it does not appear that the user is being registered correctly. When I edited the login method for logging into the Commons API so that it also logged into the Wikidata API, the value of the signed in username would be returned by the query to the WIkidata API to check the block status, but if I do not make this modification, this is the value of the user id returned. Note that I am currently checking for the block status upon creating the NearbyActivity.

![image](https://user-images.githubusercontent.com/32438079/41500467-cf4c16a4-715f-11e8-8830-96929d46f15d.png)

**Comment by seannemann21** [comment]
I know I've been gone awhile, but if anyone could give me a response on the above comment, then that'd be very helpful! @maskaravivek

**Comment by neslihanturan** [comment]
Hi @seannemann21 , I am unfamiliar with wikidata edit implementation, but @maskaravivek is not. Let's try to ping him again:) Thanks for your effort for this issue though.
